### ISSUE TITLE
source URL with IP number not recognized as source URL

### ISSUE THREAD

**Issue by sglebs** [openshift/origin#12892]
If the source repo URL has an IP address instead of hostname, OpenShift does not recognize it

##### Version

oc v1.3.3
kubernetes v1.3.0+52492b4
features: Basic-Auth

##### Steps To Reproduce

This works:
oc new-app --strategy=docker --name=gift-card-production https://gitlab.nexxera.com/gift-card/gift-card.git#dockerfile -e PORT=8080 -e PROCFILE_TARGET=web

This does not:
oc new-app --strategy=docker --name=gift-card-production https://10.9.15.192/gift-card/gift-card.git#dockerfile -e PORT=8080 -e PROCFILE_TARGET=web

Error: error: when --strategy is specified you must provide at least one source code location

##### Current Result

error: when --strategy is specified you must provide at least one source code location

##### Expected Result

```
--> Found Docker image 5533358 (13 days old) from Docker Hub for "sglebs/geventpython3"

    * An image stream will be created as "geventpython3:latest" that will track the source image
    * A Docker build using source code from https://gitlab.nexxera.com/gift-card/gift-card.git#dockerfile will be created
      * The resulting image will be pushed to image stream "gift-card-production:latest"
      * Every time "geventpython3:latest" changes a new build will be triggered
    * This image will be deployed in deployment config "gift-card-production"
```

**Comment by bparees** [comment]
this is because the cert cannot be verified when you access the site via ip, we do a git ls-remote to inspect the repo, which is failing (i've reproduced this using github.com and github's ip):

```
origin (master)$ oc new-app --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git --dry-run --loglevel=5
I0209 14:21:42.430165   32223 repository.go:367] Executing  git ls-remote https://192.30.253.113/openshift/ruby-hello-world.git
I0209 14:21:42.577942   32223 repository.go:435] Error executing command: exit status 128
F0209 14:21:42.579966   32223 helpers.go:116] error: when --strategy is specified you must provide at least one source code location
```

running the git ls-remote command manually, we see:
```
origin (master)$ git ls-remote https://192.30.253.113/openshift/ruby-hello-world.git
fatal: unable to access 'https://192.30.253.113/openshift/ruby-hello-world.git/': Unable to communicate securely with peer: requested domain name does not match the server's certificate.
```

if you want to work around this, you can set GIT_SSL_NO_VERIFY=true (but i would not recommend that).

We will see if we can improve the error message though.

**Comment by stevekuznetsov** [comment]
Similar issue:
```
Running test/cmd/newapp.sh:20: executing 'oc new-app https://github.com/openshift/rails-ex --strategy=source --dry-run' expecting success...
FAILURE after 30.173s: test/cmd/newapp.sh:20: executing 'oc new-app https://github.com/openshift/rails-ex --strategy=source --dry-run' expecting success: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
error: when --strategy is specified you must provide at least one source code location
[ERROR] PID 27188: hack/lib/cmd.sh:242: `return "${return_code}"` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/lib/cmd.sh:242: `return "${return_code}"`
[INFO] 		  2: test/cmd/newapp.sh:20: os::cmd::expect_success
[INFO]   Exiting with code 1.
```

as seen [here](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check_future/39/consoleFull#-191311657756c25df4e4b05b79524e5d52)

**Comment by bparees** [comment]
(to be clear, @stevekuznetsov means "similar in that the error output when we could not reach the repo the user provided, is not obvious as to what went wrong")

**Comment by bparees** [comment]
while fixing this we should also address this similarly confusing scenario:
https://github.com/openshift/origin/issues/12917

**Comment by bparees** [comment]
(note that I closed #12917 as a dupe of this. it hasn't actually be resolved).
